### PR TITLE
doc: add checklist for contributor's agreement

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description

Add checklist in PR template to remind checking for explicit contributor's agreement acknowledgement before merging.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [x] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -19,3 +19,4 @@ particularly focused on _why_ the work is necessary, not _what_ you did.
 * [ ] Passes all tests
 * [ ] Add comments such that someone without knowledge of the code could understand
 * [ ] Add Devel Docs in the `doc/` directory for any new code design
+* [ ] For external authors, check contributors' agreement and explict comment for ackowledgement.
```
Comment (pavanbalaji) [inline .github/pull_request_template.md]:
```suggestion
* [ ] You or your company has a signed contributor's agreement on file with Argonne
* [ ] For non-Argonne authors, request an explicit comment from your companies PR approval manager
```

Comment (hzhou) [reply .github/pull_request_template.md]: I'll make the change.

Do you mind I remove the checkbox for "Add Devel Docs in the doc/..."?

Comment (pavanbalaji) [reply .github/pull_request_template.md]: Sure.

**Comment by hzhou** [comment]
Forgot to merge this PR. Just rebased to master. @pavanbalaji Could you re-approve?

**Pull request merged**